src/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs

```diff
@@ -61,6 +61,62 @@ private bool DoHandshakeWithOptions(SslStream clientSslStream, SslStream serverS
         }
 
         [Fact]
+        public void SslStream_StreamToStream_ClientCancellation_Throws()
+        {
+            VirtualNetwork network = new VirtualNetwork();
+            using (var clientStream = new VirtualNetworkStream(network, false))
+            using (var serverStream = new VirtualNetworkStream(network, true))
+            using (var client = new SslStream(clientStream))
+            using (var server = new SslStream(serverStream))
+            using (X509Certificate2 certificate = Configuration.Certificates.GetServerCertificate())
+            {
+                SslClientAuthenticationOptions clientOptions = new SslClientAuthenticationOptions();
+                clientOptions.RemoteCertificateValidationCallback = AllowAnyServerCertificate;
+                clientOptions.TargetHost = certificate.GetNameInfo(X509NameType.SimpleName, false);
+
+                SslServerAuthenticationOptions serverOptions = new SslServerAuthenticationOptions();
+                serverOptions.ServerCertificate = certificate;
+                serverOptions.RemoteCertificateValidationCallback = AllowAnyServerCertificate;
+
+                CancellationTokenSource cts = new CancellationTokenSource();
+                Task.Run(() => cts.Cancel());
```

Comment:
What's the purpose of invoking Cancel in a Task?  This is introducing a race condition such that we won't always be testing the same thing.  If the goal is to have a canceled token before we call the operations, you can either just call cts.Cancel() synchronously or just use `new CancellationToken(true)`.  If the goal is to have cancellation occur once the operation is in progress, it'd be better to make the cts.Cancel() call synchronously below after you've got the clientTask and serverTask; that way, you know the cancellation request is coming in after all of the synchronous work done by those methods while they're actually waiting on IO.